### PR TITLE
[WIP] Build targets and CI for MS SQL on Linux

### PR DESCRIPTION
**Pull request by AlekseyMartynov**
Hi @AlekSi 
I managed to run reform CI with MS SQL on Linux.
Sharing my results.

### CODE REVIEW

**Comment by AlekSi** [comment]
Hi. Thanks for doing it. Wow, Docker image for MS SQL Server!
Unfortunately, we (well, I 😄) decided to migrate to self-hosted Drone CI instance (#63). May you use that image with Drone?

**Comment by AlekseyMartynov** [comment]
Alright. I guess if I get rid of using `sqlcmd` binary, it will be easy to test MS SQL on Drone.
Will update this PR later.

**Comment by AlekseyMartynov** [comment]
Well... go-tip failures arrived just in time 😄 
https://travis-ci.org/go-reform/reform/builds/184044736

Not sure what to do about them.

**Comment by AlekSi** [comment]
That's due to https://github.com/denisenkom/go-mssqldb/issues/193.

**Comment by AlekSi** [comment]
You can run/debug Drone builds locally: https://github.com/go-reform/reform/blob/v1-stable/.github/CONTRIBUTING.md#drone

**Comment by AlekseyMartynov** [comment]
> ERROR: This machine must have at least 3.25 gigabytes of memory to install Microsoft(R) SQL Server(R).

😬

**Comment by AlekseyMartynov** [comment]
Tried on my own Drone deployment.
Tests pass.
[drone-log.pdf](https://github.com/go-reform/reform/files/653969/drone-log.pdf)

File: .travis.yml [hunk 1]
```diff
@@ -19,7 +21,11 @@ go:
 before_install:
   - if [[ $TARGET == *mysql* ]] ; then mysql_tzinfo_to_sql /usr/share/zoneinfo | mysql -u root mysql; fi
 
-  - mkdir $GOPATH/src/gopkg.in && mv $GOPATH/src/github.com/go-reform/reform $GOPATH/src/gopkg.in/reform.v1
+  - if [[ $TARGET == *mssql-linux* ]] ; then
+      docker run -e 'ACCEPT_EULA=Y' -e 'SA_PASSWORD=Password12!' -p 1433:1433 -d microsoft/mssql-server-linux;
+    fi
+
+  - mkdir $GOPATH/src/gopkg.in && mv $GOPATH/src/github.com/$TRAVIS_REPO_SLUG $GOPATH/src/gopkg.in/reform.v1
```
Comment (AlekSi) [inline .travis.yml]: Let's revert .travis.yml changes.

File: .drone.bak [hunk 2]
```diff
@@ -0,0 +1,203 @@
+workspace:
```
Comment (AlekSi) [inline .drone.bak]: Is it backup? :) Let's remove it.

File: .drone.yml [hunk 3]
```diff
@@ -52,152 +62,7 @@ services:
...
-        REFORM_INIT_SOURCE: "root@/reform-database?parseTime=true&strict=true&sql_notes=false&time_zone='UTC'&multiStatements=true",
-        REFORM_TEST_SOURCE: "reform-user:reform-password@/reform-database?parseTime=true&strict=true&sql_notes=false&time_zone='America%2FNew_York'"
-      }
-    - {
-        GO: "captncraig/go-tip", DATABASE: sqlite3, VERSION: dummy, REFORM_DRIVER: sqlite3,
-        REFORM_INIT_SOURCE: "/tmp/reform-test.sqlite3",
-        REFORM_TEST_SOURCE: "/tmp/reform-test.sqlite3"
+        GO: "golang:1.7", DATABASE: mssql, VERSION: latest, REFORM_DRIVER: mssql,
+        REFORM_INIT_SOURCE: "server=localhost;user id=sa;password=Password12!;database=tempdb",
+        REFORM_TEST_SOURCE: "server=localhost;user id=sa;password=Password12!;database=tempdb"
```
Comment (AlekSi) [inline .drone.yml]: Please add it to [.github/drone-gen.go](https://github.com/go-reform/reform/blob/v1-stable/.github/drone-gen.go#L33).

**Comment by AlekSi** [comment]
Thanks again for doing it.

Did you make your own full Drone CI setup? `drone exec` (`make drone`) did not work?

I'm going to move build agent to https://www.scaleway.com or some other cheap hosting and add enough memory for MS SQL.

**Comment by AlekseyMartynov** [comment]
> Did you make your own full Drone CI setup?

I just exploited a small Drone 0.5 cluster which I use for one of the projects 😎
For the final PR I update both drone and drone-local yamls.

**Comment by AlekseyMartynov** [comment]
I minimized the changeset.

P.S. 
You might be aware of the bug https://github.com/drone/drone/issues/1817
I faced it here too, during the execution of MS SQL matrix item.
In short, if you see `drone exec` hanging at "[build] exit code 0", then simply use Docker version any other than 1.12.x

**Comment by AlekSi** [comment]
Rebased at #81.

**Pull request closed without merging**